src/components/Converters/ClassConverter.ts

```diff
@@ -47,14 +47,15 @@ export class ClassConverter extends RecordConverter {
         return classExportModel;
     }
 
-    protected extractImports(): string[] {
+    protected extractImports(data: RecordType): string[] {
         const rows: string[] = [];
+        const dirsUp: number = data.namespace.split(".").length;
 
         rows.push(`// tslint:disable`);
-        rows.push(`import {BaseAvroRecord} from "../../BaseAvroRecord";`);
+        rows.push(`import { BaseAvroRecord } from "` + "../".repeat(dirsUp) + `BaseAvroRecord";`);
 
         for (const enumFile of this.enumExports) {
-            const importLine = `import {${enumFile.name}} from "./${enumFile.name}Enum.ts";`;
+            const importLine = `import { ${enumFile.name} } from "./${enumFile.name}Enum";`;
```

Comment:
My compiler's blowing up on me for the explicit `.ts` suffix here, not sure if this is intended

Reply:
Was a mistake, good catch